### PR TITLE
Fix race condition when resolving performanceTiming.loadEventEnd

### PR DESCRIPTION
**Pull request by zhouyx**

### CODE REVIEW

File: src/service/variable-source.js [hunk 1]
```diff
@@ -41,7 +42,16 @@ let ReplacementDef;
  * @return {!Promise<ResolverReturnDef>}
  */
 export function getTimingDataAsync(win, startEvent, endEvent) {
-  return loadPromise(win).then(() => {
+  const deferred = new Deferred();
+  loadPromise(win).then(() => {
```
Comment (lannka) [inline src/service/variable-source.js]: would the following work?
```js
loadPromise(win)
    .then(timer.promise(1))
    .then(...);
```

Comment (zhouyx) [reply src/service/variable-source.js]: That's should work, as long as there's a `setTimeout` call involved. But do you think it's worth retrieving the `Timer` service here?

**Comment by zhouyx** [comment]
I couldn't reproduce the same race condition with our integration test fixture. Let me know if you still feel it's worth adding that.

**Comment by lannka** [comment]
Since you already have it, let's get it in. I think there is still value for it.

File: src/service/variable-source.js (hunk 1 above)
Comment (lannka) [reply src/service/variable-source.js]: Yep, the whole point of timer service is to share code.

Comment (zhouyx) [reply src/service/variable-source.js]: BTW: I found a `delayAfterDeferringToEventLoop` method from the lightbox-gallery component and few use case of `timer.promise()`. May worth introducing a global helper method for this. But I'll stick to timer.promise here.

Comment (zhouyx) [reply src/service/variable-source.js]: done

**Comment by zhouyx** [comment]
integration test added

**Comment by lannka** [comment]
Thanks for the fix!

**Pull request merged**